lib/diplomat/plug/ensure_permitted.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Diplomat.Plug.EnsurePermitted do
+  @moduledoc """
+  """
+
+  import Plug.Conn
+
+  @not_permitted "you are not permitted to perform the requested action"
+
+  @doc false
+  def init(opts), do: opts
+
+  @doc """
+  Verify against the policy that the actor is permitted to interact
+  with the given resource.
+  """
+  def call(conn, opts) do
+    policy = Keyword.fetch!(opts, :policy)
+
+    conn
+    |> current_actor
+    |> current_resource
+    |> permitted?(policy)
+    |> send_response
+  end
+
+  defp action_name(conn), do: conn.private[:phoenix_action] || {conn.method, conn.request_path}
+
+  defp current_actor(conn) do
+    actor = Diplomat.Plug.current_actor(conn)
```

Comment:
Feel like this could probably be something that could implement a behaviour. If I could configure this as a module with callbacks, I could implement Guardian or something else that knew how to get the resource from the connection